### PR TITLE
Implemented PSR12 according to PR #4533

### PR DESCRIPTION
**Pull request by catlauraherzog**
Pull Request #4533 stated to update the Coding Standard to PSR12. I implemented it within the PHP CS Fixer. If anything needs to be changed, let me know.

### CODE REVIEW

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 1]
```diff
@@ -19,13 +19,13 @@ $finder = PhpCsFixer\Finder::create()
 
 return (new PhpCsFixer\Config())
 	->setRules([
-        '@PSR2' => true,
-        'strict_param' => false,
+        '@PSR12' => true,
+        'strict_param' => true,
```
Comment (klees) [inline CI/PHP-CS-Fixer/code-format.php_cs:23]: Why did you change that?

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 2]
```diff
@@ -19,13 +19,13 @@ $finder = PhpCsFixer\Finder::create()
...
         'cast_spaces' => true,
         'concat_space' => ['spacing' => 'one'],
         'unary_operator_spaces' => true,
         'function_typehint_space' => true,
-        'return_type_declaration' => ['space_before' => 'one'],
+        'return_type_declaration' => ['space_before' => 'none'],
```
Comment (klees) [inline CI/PHP-CS-Fixer/code-format.php_cs:28]: I think we can drop this, it is already included in @PSR12 AFAIK.

**Review by klees — CHANGES_REQUESTED** [review]
@lauraherzog Thanks a lot, I left two remarks.

**Comment by mjansenDatabay** [comment]
Hi @lauraherzog and @klees,

thx a lot for this PR. IMO we need some further changes for the PSR-12 "migration".

---------

- [ ] The `PSR-2 and additions` chapter in `docs/development/coding-style.md` has to be modfied:

>
> https://github.com/ILIAS-eLearning/ILIAS/blob/2488641ea826ae8001feae5a7459b3e9cfe65a0b/docs/development/coding-style.md?plain=1#L5
>

---------

- [ ] The `Server Side (PHP)` chapter in `docs/development/README.md` has to be modified:

>
> https://github.com/ILIAS-eLearning/ILIAS/blob/2488641ea826ae8001feae5a7459b3e9cfe65a0b/docs/development/README.md#L27
>

---------

- [ ] The `PHP-CS-FIXER` examples have to be modified:

>
> https://github.com/ILIAS-eLearning/ILIAS/tree/trunk/CI/PHP-CS-Fixer/example
>

---------

- [ ] Our `PhpStorm` inspection profile in `docs/development/inspection-configs/php-storm-php-inspections.xml` has to be modified:

>
> https://github.com/ILIAS-eLearning/ILIAS/blob/2488641ea826ae8001feae5a7459b3e9cfe65a0b/docs/development/code-style-configs/php-storm.xml#L1
>

---------

- [ ] The `PhpStorm` code style config in `docs/development/code-style-configs/php-storm.xml` has to be modified:

>
> https://github.com/ILIAS-eLearning/ILIAS/blob/2488641ea826ae8001feae5a7459b3e9cfe65a0b/docs/development/inspection-configs/php-storm-php-inspections.xml#L1
>

---------

If you need any assistance, don't hesitate to contact me.

Best regards,
@mjansenDatabay

**Comment by klees** [comment]
@mjansenDatabay

* For the docu (first item in your list) we'll just have to merge #4533 before. No need to change that here, IMO.
* Could you be a bit more specific regarding the required change in the PHPStorm inspection config?

**Comment by mjansenDatabay** [comment]
> Could you be a bit more specific regarding the required change in the PHPStorm inspection config?

We'll have to import the current config/profiles files into `PhpStorm` and then change the internal coding style to `PSR-12`. Then we'll have to export these modified files again.

I'll suggest the `PhpStorm` related changes with https://github.com/ILIAS-eLearning/ILIAS/pull/4886

**Comment by klees** [comment]
Superseeded by #4960

**Pull request closed without merging**